qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -217,8 +217,18 @@ def run(self, dag):
             self.routing_pass.fake_run = False
             return dag
         # Combined
+        if self.target is not None:
+            # This is a special case SABRE only works with a bidirectional coupling graph
+            # which we explicitly can't create from the target. So do this manually here
+            # to avoid altering the shared state with the unfiltered indices.
+            target = self.target.build_coupling_map(filter_idle_qubits=True)
```

Comment:
Should `run_pass_over_connected_components` instead have a default argument `symmetric=False` so we can encapsulate this within?

Reply:
I can, but this only would ever be used in the case of a `Target` and not with a coupling map (because it's needed as symmetric earlier). I'm also not sure if there is a use case outside of sabre, it's just an implementation detail of the pass where it needs bidirectional edges for everything.

Reply:
Sounds fine to leave as is, then. Thanks!